packages/labs/task/src/task.ts

```diff
@@ -102,71 +113,92 @@ export class Task<T extends [...unknown[]] = any, R = any> {
    * is kept and only resolved when the new run is completed.
    */
   taskComplete!: Promise<R>;
+
+  /**
+   * Controls if they task will run when its arguments change. Defaults to true.
+   */
+  autoRun = true;
+
   private _resolveTaskComplete!: (value: R) => void;
   private _rejectTaskComplete!: (e: unknown) => void;
 
   constructor(
     host: ReactiveControllerHost,
-    task: TaskFunction<T, R>,
-    getDependencies: DepsFunction<T>
+    task: TaskFunction<T, R> | TaskConfig<T, R>,
+    args?: ArgsFunction<T>
   ) {
     this._host = host;
     this._host.addController(this);
+    if (typeof task === 'object') {
+      const taskConfig = task as TaskConfig<T, R>;
+      args = taskConfig.args;
+      task = taskConfig.task;
+      if (taskConfig.autoRun !== undefined) {
+        this.autoRun = taskConfig.autoRun;
+      }
+    }
     this._task = task;
-    this._getDependencies = getDependencies;
+    this._getArgs = args;
     this.taskComplete = new Promise((res, rej) => {
       this._resolveTaskComplete = res;
       this._rejectTaskComplete = rej;
     });
   }
 
   hostUpdated() {
-    this._completeTask();
+    this.performTask();
   }
 
-  private async _completeTask() {
-    const deps = this._getDependencies();
-    if (this._isDirty(deps)) {
-      if (
-        this.status === TaskStatus.COMPLETE ||
-        this.status === TaskStatus.ERROR
-      ) {
-        this.taskComplete = new Promise((res, rej) => {
-          this._resolveTaskComplete = res;
-          this._rejectTaskComplete = rej;
-        });
-      }
-      this.status = TaskStatus.PENDING;
-      this._error = undefined;
-      this._value = undefined;
-      let result!: R | typeof initialState;
-      let error: unknown;
-      // Request an update to report pending state.
-      this._host.requestUpdate();
-      const key = ++this._callId;
-      try {
-        result = await this._task(deps);
-      } catch (e) {
-        error = e;
-      }
-      // If this is the most recent task call, process this value.
-      if (this._callId === key) {
-        if (result === initialState) {
-          this.status = TaskStatus.INITIAL;
+  protected async performTask() {
+    const args = this._getArgs?.();
+    const argsDirty = this._argsDirty(args);
+    if (this.autoRun && argsDirty) {
+      this.run(args);
+    }
+  }
+
+  async run(args?: T) {
```

Comment:
oh, nice to include `args` here. Should document that it defaults to the args fn.

Reply:
Add jsdoc

Reply:
Should there be an option that controls whether the task is always run here, canceling a pending task, or if it is a no-op for a pending task? Similar to `shouldRun` in the auto-run case I guess.

Reply:
Doc added. Let's consider those other options in a follow-on issue.